src/button/styled-components.js

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+import {styled} from '../styles';
+import {KIND, SIZE, SHAPE} from './constants';
+
+export const BaseButton = styled(
+  'button',
+  ({$theme, $size, $kind, $shape}) => ({
+    ...$theme.typography[$size === SIZE.compact ? 'font200' : 'font300'],
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    border: 'none',
+    borderRadius: $shape === SHAPE.round ? '50%' : $theme.borders.radius200,
+    outline: 'none',
+    WebkitAppearance: 'none',
+    transitionProperty: 'background',
+    transitionDuration: $theme.animation.timing100,
+    transitionTimingFunction: $theme.animation.easeOutCurve,
+    cursor: 'pointer',
+    ':disabled': {
+      cursor: 'not-allowed',
+      backgroundColor: $theme.colors.mono300,
+      color: $theme.colors.mono600,
+    },
+    // Padding
+    ...($shape === SHAPE.round || $shape === SHAPE.square
+      ? {
+          paddingTop:
+            $theme.sizing[$size === SIZE.compact ? 'scale400' : 'scale500'],
+          paddingBottom:
+            $theme.sizing[$size === SIZE.compact ? 'scale400' : 'scale500'],
+          paddingLeft:
+            $theme.sizing[$size === SIZE.compact ? 'scale400' : 'scale500'],
+          paddingRight:
+            $theme.sizing[$size === SIZE.compact ? 'scale400' : 'scale500'],
+        }
+      : {
+          paddingTop:
+            $theme.sizing[$size === SIZE.compact ? 'scale200' : 'scale300'],
+          paddingBottom:
+            $theme.sizing[$size === SIZE.compact ? 'scale200' : 'scale300'],
+          paddingLeft: $theme.sizing.scale600,
+          paddingRight: $theme.sizing.scale600,
+        }),
```

Comment:
![image](https://user-images.githubusercontent.com/875591/45234233-f413c480-b289-11e8-8490-ec221ec7da58.png)

The padding feels a bit off here compared to the designs. I suppose in the designs the buttons of a minimum with, but it still feels like theres too much top/bottom padding compared to left and right. Can we have lily take a pass at these styles when she's back next week?

Reply:
Yeah definitely needs a design review. I can also tighten some of the padding here first, will take a pass through.